### PR TITLE
add base sonar integration

### PR DESCRIPTION
**Pull request by rgildein**
Add base sonar integration. 
I also added it to the [bootstack-charms-spec](https://code.launchpad.net/~rgildein/bootstack-charms-spec/+git/bootstack-charms-spec/+merge/438698).

### CODE REVIEW

**Review by esunar — CHANGES_REQUESTED** [review]

File: .github/workflows/test-sonar.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+name: SonarCloud
```
[the remark this answers is not in the document]
Comment (rgildein) [reply .github/workflows/test-sonar.yaml:1]: Do we have such .md file? Would it not better to have these files in `bootstack-charms-spec`?

[the remark this answers is not in the document]
Comment (esunar) [reply .github/workflows/test-sonar.yaml:1]: https://pastebin.canonical.com/p/FGX2w3ymKR/

[the remark this answers is not in the document]
Comment (rgildein) [reply .github/workflows/test-sonar.yaml:1]: I updated it little bit and added it to README.md.

**Review by esunar — APPROVED** [review]

**Review by Pjack — APPROVED** [review]
sonar.yaml is copied from https://github.com/canonical/charm-prometheus-juju-exporter/blob/main/.github/workflows/sonar.yml, so I'm good with it and skip the review.

About the README.md, I am thinking that we should put the document into google drive share folder. Some diagram and snapshots are useful. But right now, simple description here is also good.

**Pull request merged**